zap/src/parser/reports.rs

```diff
@@ -298,6 +310,12 @@ impl<'src> Report<'src> {
 				"this is an unbounded recursive type".to_string(),
 				"unbounded recursive types cause infinite loops".to_string(),
 			]),
+			Self::AnalyzeMissingOptValue {
+				expected,
+				required_when: required_by,
+			} => Some(vec![format!(
+				"the {expected} option should not be empty if {required_by}"
+			)]),
```

Comment:
keep the name as `required_when`, no need to change it to `required_by`

Reply:
Oh that was when I renamed the var

Reply:
```suggestion
			Self::AnalyzeMissingOptValue {
				expected,
				required_when,
			} => Some(vec![format!(
				"the {expected} option should not be empty if {required_when}"
			)]),
```